composer.json

```diff
@@ -32,16 +35,16 @@
         "php": "^7.3 || ~8.0.0",
         "laminas-api-tools/api-tools": "^1.3",
         "laminas-api-tools/api-tools-rest": "^1.3.2",
-        "laminas/laminas-hydrator": "^2.4.2 || ^3.0",
+        "laminas/laminas-hydrator": "^2.4.2 || ^3.0 || ^4.0",
         "laminas/laminas-stdlib": "^3.3",
         "laminas/laminas-view": "^2.11.3",
         "laminas/laminas-zendframework-bridge": "^1.0",
-        "phpro/zf-doctrine-hydration-module": "^2.0.1 || ^3.0 || ^4.1"
+        "api-skeletons/doctrine-orm-hydration-module": "^1.0"
     },
     "require-dev": {
-        "doctrine/doctrine-module": "^1.2.0 || ^2.1.7",
+        "doctrine/doctrine-module": "^4.0 || ^5.1",
```

Comment:
We should test only `^5.1`, and add `<5.1` to `conflict:`

Reply:
@Ocramius I didn't dropped 4.0 because we have this dependency:

https://github.com/API-Skeletons/doctrine-orm-hydration-module/blob/master/composer.json#L28

And this external module won't allow us to use only 5.1 version.

Reply:
So we aren't testing `^5.1`? Then we shouldn't include `^5.1` in our supported dependency range.

Reply:
Maybe I should do fix there? I forked this module and it works well with ^5.1.

Reply:
Ok, so here's what is not clear to me:

 * is `^4.0` required by `laminas-api-tools/api-tools-doctrine`? If not, it should be dropped
 * if `^5.1` works, we should use that, and **only** that, and drop all older versions. Introducing support for older versions should be discussed in a separate patch.

Reply:
^5.1 works with `laminas-api-tools/api-tools-doctrine` and `API-Skeletons/doctrine-orm-hydration-module`. But as for now `API-Skeletons/doctrine-orm-hydration-module` does not have ^5.1 I require - only 4.0.

And another problem is that ODM tests will be failing. The problem is here: https://github.com/laminas-api-tools/api-tools-doctrine/blob/675e8eed316b2d35d76688ed1ed1ba8bda8f1fea/test/config/application.config.php#L9. This module is abandoned, so I replaced it with `API-Skeletons/doctrine-orm-hydration-module`.

Sorry, I am new to contributing to Laminas, I am all in to do any required updates, but need help with a way to do it right.

Reply:
@tasselchof API Skeletons is my org.  If you need an update on the hydration module in support of your changes here, that can be arranged.

Reply:
@Ocramius what is the best way further? Now it works fine with ORM, but here is the deal:

```
 Problem 1
    - doctrine/doctrine-mongo-odm-module[0.11.0, 1.0.0] require doctrine/doctrine-module ^1.2 -> found doctrine/doctrine-module[1.2.0] but it conflicts with your root composer.json require (^5.1).
    - doctrine/doctrine-mongo-odm-module 1.1.0 requires doctrine/doctrine-module ^1.2 || 2.1.7 -> found doctrine/doctrine-module[1.2.0, 2.1.7] but it conflicts with your root composer.json require (^5.1).
    - doctrine/doctrine-mongo-odm-module 1.1.1 requires doctrine/doctrine-module ^1.2 || ^2.1.7 -> found doctrine/doctrine-module[1.2.0, 2.1.7, 2.1.8, 2.1.9, 2.1.10] but it conflicts with your root composer.json require (^5.1).
    - doctrine/doctrine-mongo-odm-module[2.0.0, ..., 2.0.2] require doctrine/doctrine-module ^4.0 -> found doctrine/doctrine-module[4.0.0, ..., 4.4.1] but it conflicts with your root composer.json require (^5.1).
    - Root composer.json requires doctrine/doctrine-mongo-odm-module ^0.11.0 || ^1.0.0 || ^2.0.0 -> satisfiable by doctrine/doctrine-mongo-odm-module[0.11.0, 1.0.0, 1.1.0, 1.1.1, 2.0.0, 2.0.1, 2.0.2].
```

Reply:
@TomHAnderson hydration module is tested on php 8.0?

Reply:
@TomHAnderson I've did it: https://github.com/API-Skeletons/doctrine-orm-hydration-module/pull/6.